flang/test/Lower/arrexp.f90

```diff
@@ -0,0 +1,204 @@
+! RUN: bbc -emit-fir %s -o - | FileCheck %s
+
+! CHECK-LINE: func @_QPtest1
+subroutine test1(a,b,c,n)
+  integer :: n
+  real, intent(out) :: a(n)
+  real, intent(in) :: b(n), c(n)
+  ! CHECK-DAG: %[[A:.*]] = fir.array_load %arg0(%
+  ! CHECK-DAG: %[[B:.*]] = fir.array_load %arg1(%
+  ! CHECK-DAG: %[[C:.*]] = fir.array_load %arg2(%
+  ! CHECK: %[[T:.*]] = fir.do_loop
+  ! CHECK-DAG: %[[Bi:.*]] = fir.array_fetch %[[B]]
+  ! CHECK-DAG: %[[Ci:.*]] = fir.array_fetch %[[C]]
+  ! CHECK: %[[rv:.*]] = fir.addf %[[Bi]], %[[Ci]]
+  ! CHECK: fir.array_update %{{.*}}, %[[rv]], %
+  a = b + c
```

Comment:
Would an example with more than 1 combinable elemental operation (like  `a =  b + c + d`) work ? Looking at the code, I think it should work an not generate any temp. If so, that might be worth showing this here since that will help people understanding what is happening with array expressions.